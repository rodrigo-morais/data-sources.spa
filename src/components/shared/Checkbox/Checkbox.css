.checkbox {
  display: inline-block;
  width: 10%;
  height: 30%;
  border: solid 1px #C7C5C5;
  border-radius: 50%;
  color: #ffff;
  cursor: pointer;
}

.checkbox span {
  display: inline-block;
  width: 100%;
  text-align: center;
  padding-top: 10%;
}

.checkbox--checked {
  background-color: #49E5CD;
}

.label {
  margin: 0 0 0 1%;
  position: absolute;
  display: inline-block;
  color: #000000;
  font-size: 12px;
  font-weight: 500;
  cursor: pointer;
}
