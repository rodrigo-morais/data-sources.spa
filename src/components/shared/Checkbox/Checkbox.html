<div @click="handleToggle" v-bind:class="[className]">
  <div v-bind:class="[checkboxClass, {'checkbox--checked': checked}]">
    <span>V</span>
  </div>
  <p class="label">{{label}}</p>
</div>
