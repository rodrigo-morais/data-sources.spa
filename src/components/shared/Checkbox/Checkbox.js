import Vue from 'vue'
import Component from 'vue-class-component'

export default Component({
  name: 'Checkbox',
  props: {
    className: { type: String, default: '' },
    label: String,
    isChecked: { type: Boolean, default: false },
  },
})(class Checkbox extends Vue {
  checked = this.isChecked
  checkboxClass = 'checkbox'

  handleToggle() {
    this.checked = !this.checked
    this.$emit('onClick', this.checked)
  }
})
