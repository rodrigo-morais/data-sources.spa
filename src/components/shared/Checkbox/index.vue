<template src='./Checkbox.html'></template>

<script src='./Checkbox.js'></script>

<style scoped src='./Checkbox.css'></style>
