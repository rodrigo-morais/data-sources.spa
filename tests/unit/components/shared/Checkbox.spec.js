import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import Checkbox from '@/components/shared/Checkbox/index.vue'

const props = {
  label: 'Select',
}

describe('Checkbox.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Checkbox)
  })

  it('renders according to design', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('renders according to design when is checked', () => {
    const wrapperChecked = shallowMount(Checkbox, {
      propsData: { ...props, isChecked: true },
    })
    const renderer = createRenderer()
    renderer.renderToString(wrapperChecked.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('calls `onClick` props method when execute `handleToggle`', () => {
    const checkbox = wrapper.find('.checkbox')
    checkbox.trigger('click')

    expect(wrapper.emitted().onClick).toBeTruthy()
    expect(wrapper.emitted().onClick[0]).toEqual([true])
  })
})
