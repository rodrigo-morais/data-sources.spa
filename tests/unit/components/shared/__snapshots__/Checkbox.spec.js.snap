// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox.vue renders according to design 1`] = `
<div>
  <div class="checkbox"><span>V</span></div>
  <p class="label"></p>
</div>
`;

exports[`Checkbox.vue renders according to design when is checked 1`] = `
<div>
  <div class="checkbox checkbox--checked"><span>V</span></div>
  <p class="label">Select</p>
</div>
`;
